taccsite_cms/contrib/taccsite_sysmon/static/taccsite_sysmon/js/sysmon.js

```diff
@@ -0,0 +1,168 @@
+
+/**
+ * Single system data
+ * @typedef {object} System
+ * @see https://frontera-portal.tacc.utexas.edu/api/system-monitor/
+ */
+
+// Allow system mointor to work(-ish) on local server
+const USE_SAMPLE_DATA = (window.location.hostname === 'localhost');
+const API_SAMPLE_DATA = JSON.parse('[{"hostname": "frontera.tacc.utexas.edu", "display_name": "Frontera", "ssh": {"type": "ssh", "status": true, "timestamp": "2021-07-30T19:45:02Z"}, "heartbeat": {"type": "heartbeat", "status": true, "timestamp": "2021-07-30T19:45:02Z"}, "status_tests": {"ssh": {"type": "ssh", "status": true, "timestamp": "2021-07-30T19:45:02.176Z"}, "heartbeat": {"type": "heartbeat", "status": true, "timestamp": "2021-07-30T19:45:02.174Z"}}, "resource_type": "compute", "jobs": {"running": 322, "queued": 1468, "other": 364}, "load_percentage": 99, "cpu_count": 472760, "cpu_used": 468616, "is_operational": true}, {"hostname": "stampede2.tacc.utexas.edu", "display_name": "Stampede2", "ssh": {"type": "ssh", "status": true, "timestamp": "2021-07-30T19:45:03Z"}, "heartbeat": {"type": "heartbeat", "status": true, "timestamp": "2021-07-30T19:45:03Z"}, "status_tests": {"heartbeat": {"type": "heartbeat", "status": true, "timestamp": "2021-07-30T19:45:03.069Z"}, "ssh": {"type": "ssh", "status": true, "timestamp": "2021-07-30T19:45:03.074Z"}}, "resource_type": "compute", "jobs": {"running": 1115, "queued": 1032, "other": 444}, "load_percentage": 96, "cpu_count": 1309056, "cpu_used": 1257184, "is_operational": true}]');
+
+/**
+ * The URL of the API endpoint
+ * @type {string}
+ */
+const API_URL = '/api/system-monitor';
+
+/**
+ * The systems to show
+ *
+ * _Notice: This value is expected to be available from another script_
+ * @type {string}
+ */
+const SYSTEM_HOSTNAME = window.SYSMON_SYSTEM_HOSTNAME;
+
+/**
+ * Load system status
+ * @param {string} path
+ * @param {function} onSuccess - Callback for success (receives JSON)
+ * @param {function} onError - Callback for success (receives XMLHttpRequest)
+ */
+function loadStatus(path, onSuccess, onError) {
+  var xhr = new XMLHttpRequest();
+  xhr.onreadystatechange = function () {
+    if (xhr.readyState === XMLHttpRequest.DONE) {
+      if (xhr.status === 200) {
+        if (onSuccess) onSuccess(JSON.parse(xhr.responseText));
+      } else {
+        if (onError) onError(xhr);
+      }
+    }
+  };
+  xhr.open('GET', path, true);
+  xhr.send();
+}
+
+/**
+ * Whether system is operational
+ * @param {System} system
+ * @return {boolean}
+ */
+function isOperational(system) {
+  if (system['load_percentage'] < 1 || system['load_percentage'] > 99) {
+    system['load_percentage'] = 0;
+    return system['jobs']['running'] > 1;
+  }
+  return true;
+}
+
+/**
+ * Show system content in UI
+ */
+function showStatus() {
+  document.getElementById('status').classList.remove('d-none');
+}
+
+/**
+ * Style system status
+ * @param {string} type - A type: "warning"
+ */
+function setStatusStyle(type) {
+  switch (type) {
+    case 'warning':
```

Comment:
Will there be more `type`s to compare against? If not, is there reason that `switch` used here (in comparison to other blocks?

Reply:
I began the work for this, but then abandoned it, becuase it grew beyond the scope of a "minimal" plugin. I have added a requirement on #295 to support multiple, distinct states (rather than assume failure and ignore loading).